sysctl.d/50-coredump.conf.in

```diff
@@ -9,4 +9,4 @@
 # and systemd-coredump(8) and core(5) for the explanation of the
 # setting below.
 
-kernel.core_pattern=|@rootlibexecdir@/systemd-coredump %P %u %g %s %t %c %h
+kernel.core_pattern=|@rootlibexecdir@/systemd-coredump %P %s %t %c %h
```

Comment:
not happy about this, this is nasty during upgrades...

I'd really prefer if we'd trust the data the kernel passes to us directly as much as we can, and only rely on /proc where that's not enough. or in other words, I'd prefer if the pattern would stay the same and we'd continue to use the data on the host, and read the data from /proc only when we need to translate it to the container

Reply:
I agree that we should be backward compat with older versions of systemd-coredump that run in containers.

OTOH the current interface of systemd-coredump doesn't allow any changes/improvements. 

Maybe we could change the interface in an incompatible manner but rather than trying to execute an old version in the container we could simply handle a crash from the host as it used to be done.

Not sure how we could figure out the version of systemd-coredump running in the container though...

Reply:
> I'd really prefer if we'd trust the data the kernel passes to us directly as much as we can, and only rely on /proc where that's not enough.

Well, it seems less error prone to only use the minimal data from the kernel and retrieve the rest from /proc. It seems much less error prone as we won't forget any metadata translations in this case. This also makes the code simpler as the handling of the metadata will be pretty similar for both cases.

Reply:
I still think we should probably prefer /proc over the command line. The kernel's command line handling is nuts anyway, since they don't understand escaping and such... If we read what we need from /proc we know exactly how to get it, in a fully past and future proof way. and we know exactly how to convert the data between namespaces correctly.

Hence still, let's just keep the line we already use and pick up the additional data from /proc, unless the data *really* isn't available there.

Reply:
I'm confused as you initially said:

> I'd really prefer if we'd trust the data the kernel passes to us directly as much as we can, and only rely on /proc where that's not enough

but now you said:

> I still think we should probably prefer /proc over the command line.

I'll go for the second option as we seem to agree now.